catroid/src/main/java/org/catrobat/catroid/embroidery/ShareEmbroideryFile.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.embroidery;
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.PointF;
+import android.net.Uri;
+import android.support.annotation.VisibleForTesting;
+import android.support.v4.content.FileProvider;
+import android.util.Log;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.Constants;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class ShareEmbroideryFile {
+	private static final String TAG = ShareEmbroideryFile.class.getSimpleName();
+
+	public ShareEmbroideryFile() {
```

Comment:
if there is only the default constructor, you don't need to put it explicitly. from wikipedia:
>"In both Java and C#, a "default constructor" refers to a nullary constructor that is automatically generated by the compiler if no constructors have been defined for the class."